pkg/validate/dynamic/dynamic.go

```diff
@@ -438,22 +424,28 @@ func (c closure) usingCheckAccessV2() (bool, error) {
 		return false, nil
 	}
 
+	actionsToFind := map[string]struct{}{}
 	for _, action := range c.actions {
-		found := false
-		for _, result := range results.Value {
-			if result.ActionId == action {
-				found = true
-				if result.AccessDecision == remotepdp.Allowed {
-					break
-				}
-				return false, nil
+		actionsToFind[action] = struct{}{}
+	}
+	for _, result := range results.Value {
+		_, ok := actionsToFind[result.ActionId]
+		if ok {
+			delete(actionsToFind, result.ActionId)
+			if result.AccessDecision == remotepdp.Allowed {
+				continue
 			}
-		}
-		if !found {
-			c.dv.log.Infof("The result didn't include permission %s", action)
 			return false, nil
 		}
 	}
+	if len(actionsToFind) > 0 {
+		missingActions := []string{}
+		for missingAction := range actionsToFind {
+			missingActions = append(missingActions, missingAction)
+		}
+		c.dv.log.Infof("The result didn't include permissions %v", missingActions)
```

Comment:
We should also log the name or object ID of the SP/identity being validated so that SREs reading the RP logs will know which identity is missing the permissions when they read this message.